plugins/by-name/ltex-extra/default.nix

```diff
@@ -44,24 +52,65 @@ lib.nixvim.plugins.mkNeovimPlugin {
   };
 
   extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
-      '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+      {
+        when = !lspCfg.enable;
+        message = ''
+          You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+          You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+        '';
+      }
+      (
+        let
+          expectedDefs = map toString [
+            ./.
+            ../../lsp/language-servers
+          ];
+          isExternal = d: !elem d.file expectedDefs;
+          anyExternal =
+            acc: name: v:
+            let
+              e = findFirst isExternal null v.definitionsWithLocations;
+            in
+            if acc != null then
+              acc
+            else if e == null then
+              null
+            else
+              {
+                inherit name;
+                inherit (e) file;
+              };
+          external = foldlAttrs anyExternal null options.plugins.lsp.servers.ltex;
```

Comment:
Interesting approach using the definition location 😎

I wonder if it'd be better to _filter_ for external definitions, instead of reducing to the _first_ match? That way we could list all offending definitions in the warning.

I'm also a little concerned about false-pisitives where there are irrelevant low-prio defs. I can't recall ottomh whether `definitions` includes _all_ definitions or only definitions of the `highestPrio`? I can check this in `nix repl` when I'm next at a PC.

Reply:
Another consideration is if/how `mkIf false` definitions show up. E.g. a user might have a toggle somewhere between a manual configuration of the lsp and using this plugin, implemented using `mkIf`.